src/btree/row_modify.c

```diff
@@ -392,8 +396,12 @@ __wt_update_obsolete_check(
      * subsequent to it, other threads of control will terminate their walk in this element. Save a
      * reference to the list we will discard, and terminate the list.
      */
-    if (first != NULL && (next = first->next) != NULL &&
-      __wt_atomic_cas_ptr(&first->next, next, NULL)) {
+    if (first != NULL && (next = first->next) != NULL) {
+        /*
+         * No need to use a compare and swap because we have obtained a lock before entering this
+         * function.
+         */
+        first->next = NULL;
         /*
```

Comment:
A blank line here would imo help readability. Feel free to disregard this if you don't agree.